File: codechef/JUNE17ES.py
from decimal import *
import sys;
from sys import *
#A solution to Euler sum published in June long challenge 2017.
getcontext().prec = 400;
e =Decimal(1);
nforexp=1;
i = Decimal(1);
for i in range(1, 1500):
  nforexp*=i;
  e+=Decimal(1)/nforexp;
#print e;
def sume(n):
  ans=(n*(n+1)/2);
  x=Decimal(e-1);
  sgn=1;
  #count=0;
  while(1):
    #count+=1;
    if(n==0):
      break;
    if(n==1):
      ans += sgn;
      break;
    m = int(n*(x-1));
    a = (n+m)*(n+m+1)/2;
    n = m;
    ans += int(sgn*a);
    sgn*=-1;
    x = 1/(1-1/x);
    while(x>2):
      ans += sgn*n*(n+1)/2;
      x -= 1;

  #print count;

  stdout.write(str(ans));
def main():
  n=int(sys.stdin.readline());
  sume(n);
  return 0;

main();

File: codechef/JUNE17ES.cpp
//A solution to Euler sum published in June long challenge 2017.
#include<iostream>
const long double e=2.7182818284, beta = e/(e-1);
using namespace std;
long long S(long long n)
{ if(n==0)
    return 0;
  /*if(n==1)
    return 4;
  if(n==2)
    return 8;*/
  long long m=n*(beta-1);
  long long a=m*n + n*(n+1)/2;
  a-=m*(m+1)/2;
  a-=S(m);
  return a;
}/*
long long sume(long long n)
{ long long m=e*n, ans=m*(m+1)/2;
  ans-=S(n*(e-1));
  return ans;
}*/
int main()
{ /*int n;
  cin>>n;
  cout<<sume(n);*/
  long long ans=0;
  for(int i=1; i<=10; i++)
  { ans+=i*beta;
    cout<<i<<' '<<ans<<' '<<S(i)<<endl;
  }
  //cout<<S(2);
  return 0;
}
